src/com/facebook/buck/jvm/java/AbstractJavacOptions.java

```diff
@@ -230,7 +233,13 @@ public void appendOptionsTo(
       allPluginsBuilder.addAll(javacPlugins);
 
       for (ResolvedJavacPluginProperties properties : javacPlugins) {
-        optionsConsumer.addFlag("Xplugin:" + properties.getProcessorNames().first());
+        // Javac plugins should be declared as:
+        // '-Xplugin:<pluginName> <arg1> <arg2> <arg3> ...'
+        String pluginName = properties.getProcessorNames().first();
+        String pluginArguments =
+            getJavacPluginArguments().getOrDefault(pluginName, ImmutableList.of()).stream()
+                .reduce("", (s, s2) -> s + " " + s2);
+        optionsConsumer.addFlag("Xplugin:" + pluginName + pluginArguments);
```

Comment:
Nit: This is a bit of a misuse of `addFlag`. You probably want to do `addFlag` followed by `addExtras`. That would also allow you to get rid of the string reducing logic above. Unless you're doing this because the JVM literally expects this whole string as one atomic arg that includes space characters, in which case, add a comment.